src/components/Scheme/SchemeContainer.tsx

```diff
@@ -8,28 +9,51 @@ import * as React from "react";
 import { BreadcrumbsItem } from "react-breadcrumbs-dynamic";
 import { Link, Route, RouteComponentProps, Switch } from "react-router-dom";
 import * as Sticky from "react-stickynode";
+import { showNotification } from "reducers/notifications";
+import { IRootState } from "reducers";
+import { connect } from "react-redux";
 import ReputationFromToken from "./ReputationFromToken";
 import SchemeInfoPage from "./SchemeInfoPage";
 import SchemeProposalsPage from "./SchemeProposalsPage";
 import * as css from "./Scheme.scss";
 
+interface IDispatchProps {
+  showNotification: typeof showNotification;
+}
 
-interface IProps {
+interface IExternalProps {
   currentAccountAddress: Address;
+  history: H.History;
 }
 
-export default class SchemeContainer extends React.Component<IProps & RouteComponentProps<any>, null> {
+type IProps = IExternalProps & IDispatchProps;
+
+const mapStateToProps = (_state: IRootState, ownProps: IExternalProps): IExternalProps => {
+  return ownProps;
+};
+
+const mapDispatchToProps = {
+  showNotification,
+};
 
-  public render() {
+class SchemeContainer extends React.Component<IProps & RouteComponentProps<any>, IExternalProps> {
+
+  public render(): any {
     const { currentAccountAddress, match } = this.props;
     const schemeId = match.params.schemeId;
     const daoAvatarAddress = match.params.daoAvatarAddress;
+    const handleNewProposal = async (e: any): Promise<void> => {
```

Comment:
i would normally put this outside of the render function, feels weird to put it inside, but not a big deal

Reply:
that is a lot easier as is so will leave it.

Reply:
so there is a reason to not put it in the render function: the render function is called very often, and so as a rule should be as 'light" as possible. (Not that a function def like this gives significant overhead, but as a general ruleof thumb, what can be done outside of the render function should be outside)

Reply:
this is so trivial a "load" I don't see any reason to worry about it.  the alternative is more complicated, and therefore risky, code.  the only place this function is used is from within render, and from that perspective it makes perfect sense to define it within, though yes, out of habit, normally I would likely put the function outside of render, as I tried to do but changed my mind as this version is simpler.

If you are insisting on doing this to approve  this PR, then I will, otherwise no.  @tibetsprague   @jellegerbrandy

Reply:
It is a line or two of code extra, but it is the standard pattern, and for the good reasons that I mentioned. (The "more risky" argument sounds absurd to me :-))